registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspCsvServiceImpl.java

```diff
@@ -87,6 +88,50 @@ public void importCsv(byte[] file) {
         aspspRepository.saveAll(uuidGeneratorService.checkAndUpdateUUID(aspsps));
     }
 
+    @Override
+    public CsvFileMergeValidationReportBO validateMergeCsv(byte[] file) {
+        List<AspspBO> input = readAllRecords(file, aspspBOConverter::csvRecordListToAspspBOList);
+        FileValidationReportBO fileValidationReport = aspspValidationService.validate(input);
+
+        List<AspspBO> database = aspspBOConverter.toAspspBOList(aspspRepository.findAll());
+        Set<AspspBO> difference = new HashSet<>();
+        int numberOfNewRecords = 0;
+
+        for (AspspBO inputItem : input) {
+            boolean matched = false;
+
+            for (AspspBO dbItem : database) {
+                if (Objects.equals(inputItem, dbItem)) {
+                    matched = true;
+                    break;
+                }
+
+                if (isAspspModified(inputItem, dbItem)) {
+                    difference.add(inputItem);
+                    matched = true;
+                    break;
+                }
```

Comment:
It seems to me, that `isAspspModified` returns *true* only if ids are different. Shouldn't it return *true* if any field differs?

Reply:
Sorry, but parentheses have confused you a little bit (yeah, I know that they are not that readable, that's why I used extra tabulation there): 
![image](https://user-images.githubusercontent.com/24653053/69338434-e0286880-0c6b-11ea-9f48-c7c5edbc5f3d.png)
As you can see, the `!()` wraps each comparison except `BIC` and `Bank Code` ones.